src/test/java/uk/gov/pay/adminusers/service/SecondFactorAuthenticatorTest.java

```diff
@@ -28,6 +37,25 @@ public void shouldGenerateAndValidate2FAPasscode() throws Exception {
         assertFalse(secondFactorAuthenticator.authorize(secret + 1, password));
     }
 
+    @Test
+    public void shouldSuccess_ifAskedToValidateLastSteps2FAPasscode() throws Exception {
```

Comment:
I would be tempted to do a test case and put an intermediate mils figure too 30seconds and one just below the max default of past 4 like 120 seconds